src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -170,8 +172,14 @@ public EmailWrapper generateFeedbackSessionSummaryOfCourse(String courseId, Stri
                     "${submitUrl}", submitUrlHtml,
                     "${reportUrl}", reportUrlHtml));
         }
+
+        if (linksFragmentValue.length() == 0) {
```

Comment:
When will this be triggered? Do we have test case for it?

Reply:
This will occur if the course does not have any feedback sessions that have sent the opening or published email.

Basically, if `sessions` is empty.
https://github.com/TEAMMATES/teammates/blob/ef506f9fb5122c2eaef5e4b99a1434329b253629/src/main/java/teammates/logic/api/EmailGenerator.java#L124-L131

Will add a test case for this